uniffi_bindgen/src/templates/macros.rs

```diff
@@ -50,14 +50,15 @@ UNIFFI_HANDLE_MAP_{{ obj.name()|upper }}.insert_with_output(err, || {
 {% endmacro %}
 
 {% macro to_rs_method_call(obj, meth) %}
+{% let mut_suffix = obj.threadsafe()|choose("", "_mut") -%}
```

Comment:
Hrm, this `mut_suffix` business seems unfortunate. I feel like we might be able to clean this up by using a custom trait, along the lines of this but with better names:

```
trait UniffiMethodCall {
  pub fn call_method_with_result(&self, err, callback) -> Result;
}

impl UniffiMethodCall for ConcurrentHandleMap {
  pub fn call_method_with_result(&self, err, callback) -> Result {
      // do the mut thing internally
      self.call_with_result_mut(...)
  }
}

impl UniffiMethodCall for NonLockingHandleMap {
  pub fn call_method_with_result(&self, err, callback) -> Result {
      // no mut here
      self.call_with_result(...)
  }
}
```

But that may be more fiddling around is really necessary, and I'd be happy to land the simpler `mut_suffix` version to get this out the door.

Reply:
This definitely looks better than changing to code that gets generated.

It's more complicated than I thought, because the callback type for the two implementations varies as `FnOnce(&T)` and `FnOnce(&mut T)`.